libethcore/Miner.h

```diff
@@ -397,7 +397,11 @@ class Miner : public Worker
     float RetrieveHashRate() noexcept { return m_hashRate.load(std::memory_order_relaxed); }
     void TriggerHashRateUpdate() noexcept
     {
-        m_hashRateUpdate.store(true, std::memory_order_relaxed);
+        bool b = false;
+        if (m_hashRateUpdate.compare_exchange_strong(b, true))
+            return;
+        // GPU didn't respond to last trigger, assume it's dead.
+        m_hashRate = 0.0;
```

Comment:
Using huge values for `--cuda-grid-size` and/or `--cuda-streams` could result that the hashrate is not updated within `m_collectInterval (Farm.h)`. So the reported hashrate would drop to 0.

Reply:
Correct, but they would have to be unreasonably large. The UpdateHashRate is called every 5 seconds.

Reply:
> Correct, but they would have to be unreasonably large. The UpdateHashRate is called every 5 seconds.

I'll add a comment when submitting my next PR.